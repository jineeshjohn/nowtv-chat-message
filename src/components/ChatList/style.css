
.container {
  border: 1px solid #000;
  background-color: #f1f1f1;
  border-radius: 5px;
  padding: 10px;
  margin: 10px 0;
  float: left;
  clear: both;
  width: 95%;
}

.container .fullname {
  font-size: 12px;
}

.container .email {
  font-size: 13px;
  color: #bbbbbb;
}

.container .email {
  font-size: 13px;
  color: #333;
  line-height: 13px;
  height: 13px;
}

.container .message {
  font-size: 12px;
  color: #4c0404;
}


.container img {
  float: left;
  max-width: 60px;
  width: 100%;
  margin-right: 20px;
}

.time-right {
  float: right;
  color: #0c096f;
  font-size: 12px;
  padding: 5px;
}

