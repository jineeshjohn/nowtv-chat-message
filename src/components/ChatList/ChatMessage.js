import React, { Component } from "react";
import { PropTypes } from "prop-types";
import "./style.css";
import defaultAvatar from "./../../resources/avatar.png";

class ChatMessage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showEmail: false
    };
    this.mouseEnter = this.mouseEnter.bind(this);
    this.mouseLeave = this.mouseLeave.bind(this);
  }
  mouseEnter() {
    this.setState({
      showEmail: true
    });
  }
  mouseLeave() {
    this.setState({
      showEmail: false
    });
  }
  render() {
    const { message, fullName, timestamp, avatar, email } = this.props;
    const { showEmail } = this.state;
    return (
      <div
        className="container"
        onMouseEnter={this.mouseEnter}
        onMouseLeave={this.mouseLeave}
      >
        <div>
          <img src={avatar || defaultAvatar} alt="Avatar" />
          <div>
            <div className="email">{showEmail && email}</div>
            <div className="fullname">{fullName}</div>
            <div className="message">{message}</div>
          </div>
        </div>
        <span className="time-right">{timestamp}</span>
      </div>
    );
  }
}

ChatMessage.propTypes = {
  messageId: PropTypes.string,
  userId: PropTypes.string,
  fullName: PropTypes.string,
  timestamp: PropTypes.string,
  email: PropTypes.string,
  message: PropTypes.string,
  avatar: PropTypes.string
};

ChatMessage.defaultProps = {
  avatar: null
};

export default ChatMessage;
