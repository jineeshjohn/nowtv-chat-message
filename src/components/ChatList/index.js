import React, { Fragment } from "react";
import { PropTypes } from "prop-types";
import "./style.css";
import ChatMessage from "./ChatMessage";

const ChatList = ({ chatMessage }) => {
  if (chatMessage.length === 0) {
    return <div>Failed to load api</div>; // failure case
  }
  const list = chatMessage.map(chat => (
    <ChatMessage key={chat.messageId} {...chat} />
  ));
  return (
    <Fragment>
      <div className="messages">{list}</div>
    </Fragment>
  );
};

ChatList.propTypes = {
  chatMessage: PropTypes.arrayOf(
    PropTypes.shape({
      messageId: PropTypes.string,
      userId: PropTypes.string,
      fullName: PropTypes.string,
      timestamp: PropTypes.string,
      email: PropTypes.string,
      message: PropTypes.string,
      avatar: PropTypes.string
    })
  )
};

ChatMessage.defaultProps = {
  chatMessage: []
};

export default ChatList;
