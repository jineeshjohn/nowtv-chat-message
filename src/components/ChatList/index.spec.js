import React from 'react';
import { shallow } from 'enzyme';
import ChatList from './index';

describe('<ChatList>', ()=>{
  const chatMessage = [{
    messageId: '12356',
    userId: '613651251',
    fullName: 'Robin Balmforth',
    timestamp: '2016-06-03T20:24:29Z',
    email: 'robin@example.com',
    message: 'Hello, World!',
    avatar: null
  },
  {
    messageId: '123567',
    userId: '6136512517',
    fullName: 'Robin Balmforthh',
    timestamp: '2016-06-03T20:24:29Zh',
    email: 'robin6@example.com',
    message: 'Hello, World!',
    avatar: null
  }];


  it('renders without crashing', () => {
    const wrapper = shallow(<ChatList chatMessage={chatMessage}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have one messages container', () => {
    const wrapper = shallow(<ChatList chatMessage={chatMessage}/>);
    expect(wrapper.find('.messages').length).toEqual(1);
  });
});