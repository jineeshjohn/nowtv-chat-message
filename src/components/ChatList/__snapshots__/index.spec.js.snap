// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList> renders without crashing 1`] = `
<Fragment>
  <div
    className="messages"
  >
    <ChatMessage
      avatar={null}
      chatMessage={Array []}
      email="robin@example.com"
      fullName="Robin Balmforth"
      key="12356"
      message="Hello, World!"
      messageId="12356"
      timestamp="2016-06-03T20:24:29Z"
      userId="613651251"
    />
    <ChatMessage
      avatar={null}
      chatMessage={Array []}
      email="robin6@example.com"
      fullName="Robin Balmforthh"
      key="123567"
      message="Hello, World!"
      messageId="123567"
      timestamp="2016-06-03T20:24:29Zh"
      userId="6136512517"
    />
  </div>
</Fragment>
`;
